decode.go

```diff
@@ -370,3 +390,42 @@ func (de *decoder) slice(slval reflect.Value, vb []byte) error {
 	}
 	return nil
 }
+
+func (de *decoder) handleMap(slval reflect.Value, vb []byte) error {
```

Comment:
Write a comment - 'implementation according to Google's protobuf-implementation' or something like that